src/CollectWidget.res

```diff
@@ -314,22 +459,142 @@ let make = (
-        | Some(pmt) => renderInputs(pmt)
-        | None => renderPMTOptions()
+  }
+
+  let handleTabSelection = selectedPMT => {
+    if availablePaymentMethodTypes->Array.indexOf(selectedPMT) >= defaultOptionsLimitInTabLayout {
+      // Move the selected payment method at the last tab position
+      let ordList = availablePaymentMethodTypes->Array.reduceWithIndex([], (acc, pmt, i) => {
+        if i === defaultOptionsLimitInTabLayout - 1 {
+          acc->Array.push(selectedPMT)
+        } else if pmt !== selectedPMT {
+          acc->Array.push(pmt)
         }
-      }}
+        acc
+      })
+      setAvailablePaymentMethodTypesOrdered(_ => ordList)
+    }
+    setSelectedPaymentMethodType(_ => Some(selectedPMT))
+  }
+
+  let renderTabScreen = (~limit=defaultOptionsLimitInTabLayout) => {
+    let activeStyles: JsxDOM.style = {
+      borderColor: primaryTheme,
+      borderWidth: "2px",
+      color: primaryTheme,
+    }
+    let defaultStyles: JsxDOM.style = {
+      borderColor: "#9A9FA8",
+      borderWidth: "1px",
+      color: primaryTheme,
+    }
+    // tabs
+    <div
+      className="flex flex-col w-full min-w-[300px] max-w-[520px]
+      lg:min-w-[400px]">
+      <div>
+        {switch savedPMD {
+        | Some(pmd) => renderFinalizeScreen(pmd)
+        | None =>
+          <div>
+            <div className="flex flex-row w-full">
+              {availablePaymentMethodTypesOrdered
+              ->Array.reduceWithIndex([], (items, pmt, i) => {
+                if i < limit {
+                  items->Array.push(
+                    <div
+                      key={i->Int.toString}
+                      onClick={_ => setSelectedPaymentMethodType(_ => Some(pmt))}
+                      className="flex w-full items-center rounded border border-solid border-jp-gray-700 px-2.5 py-1.5 mr-2.5 cursor-pointer hover:bg-jp-gray-50"
+                      style={selectedPaymentMethodType === Some(pmt)
+                        ? activeStyles
+                        : defaultStyles}>
+                      {pmt->getPaymentMethodTypeIcon}
+                      <div className="ml-2.5"> {React.string(pmt->getPaymentMethodTypeLabel)} </div>
+                    </div>,
+                  )
+                }
+                items
+              })
+              ->React.array}
+              {if availablePaymentMethodTypesOrdered->Array.length > limit {
+                <div className="relative">
+                  <Icon
+                    className="absolute z-10 pointer translate-x-2.5 translate-y-3.5 pointer-events-none"
+                    name="arrow-down"
+                    size=10
+                  />
+                  <select
+                    className="h-full relative rounded border border-solid border-jp-gray-700 py-1.5 cursor-pointer bg-white text-transparent w-8
+                    hover:bg-jp-gray-50 focus:border-0.5">
+                    {switch selectedPaymentMethodType {
```

Comment:
Render if can be used
          <RenderIf condition={selectedPaymentMethodType->Option.isSome}>

Reply:
we need the `selectedPaymentMethodType`, which is an option, so will need to map or get the value using a default,

```

<RenderIf condition={selectedPaymentMethodType->Option.isSome}>
  <option
    value={selectedPaymentMethodType->Option.map(getPaymentMethodTypeLabel)->Option.getOr("")}
    disabled={true}>
    {React.string(selectedPaymentMethodType->Option.map(getPaymentMethodTypeLabel)->Option.getOr(""))}
  </option>
</RenderIf>
```

doesn't look very good

Reply:
got it